### PR TITLE
Ratvar-converted AIs become brass-colored, speak in Ratvarian, and cannot be carded

### PR DESCRIPTION
**Pull request by ChangelingRain**
:cl: Joan
rscadd: Ratvar-converted AIs become brass-colored, speak in Ratvarian, and cannot be carded.
/:cl:

wow woah
Effectively a way to make it really fucking obvious that the clockcult A: has the AI and B: is probably setting up there.

### CODE REVIEW

**Comment by GunHog** [comment]
What is the reasoning behind this change? What prompted it?

**Comment by Cyberboss** [comment]
#23652 prompted this. Ban kneejerk PRs when?

**Comment by ChangelingRain** [comment]
Another oldman rant; basically using the AI sat as a base is too easy and converting the AI isn't noticeable enough, especially with how fast you can do it.

**Comment by ChangelingRain** [comment]
And yes @Cyberboss that pr prompted this that doesn't mean it's fucking wrong you dingbat.

**Comment by francinum** [comment]
AI takes 10 years to respond, shoves everything through rot13 as to not show itself.

**Comment by RemieRichards** [comment]
@francinum they won't be able to remove the graves, accents and hyphens that are also automatically added to ratvarian, so even if they did have rot13 on hand they'd give themselves away.

**Comment by francinum** [comment]
It's however massively less likely, as quirks like that are calculated pre-rot13.

**Comment by RemieRichards** [comment]
@francinum they're no less likely though, a grave, hyphen or accent just get ignored by rot13, only alphabetical characters are actually rotated.

**Comment by francinum** [comment]
Let me pull the code and show what I mean.

**Comment by RemieRichards** [comment]
You mean the rot13/ratvarian code? cos I don't think pulling it out against me, it's author, will help your cause :P

**Comment by ChangelingRain** [comment]
He's unfortunately correct; all of the matches won't catch text that's already ratvarian.

**Comment by ChangelingRain** [comment]
However this is super fucking easy to fix.

**Comment by RemieRichards** [comment]
Easy fix yeah. just run the regexes through rot13 lmao.

**Comment by ChangelingRain** [comment]
no easier than that just run the regexes again after you translate; either way you're going to be running them twice either way, and you don't even need to duplicate them for this

File: code/game/gamemodes/clock_cult/clock_helpers/ratvarian_language.dm [hunk 1]
```diff
@@ -77,18 +77,34 @@ List of nuances:
 	ratvarian 		= replacetext(ratvarian, 	RATVAR_POST_AND_MATCH,	RATVAR_POST_AND_REPLACEMENT)
 	ratvarian 		= replacetext(ratvarian, 	RATVAR_TO_MATCH, 		RATVAR_TO_REPLACEMENT)
 	ratvarian 		= replacetext(ratvarian, 	RATVAR_MY_MATCH, 		RATVAR_MY_REPLACEMENT)
-	return rot13(ratvarian)
+	ratvarian 		= rot13(ratvarian) //run the regexes twice, so that you catch people translating it beforehand
```
Comment (RemieRichards) [inline code/game/gamemodes/clock_cult/clock_helpers/ratvarian_language.dm]: put the regexes into a proc, and run it twice.

**Comment by Robustin** [comment]
I don't think this will change the endgame meta, subverting the AI and turning the satellite into an impossible array of defenses will still be the go-to for all ark summoning.

However, this is still a good change because it at least forces the clock cult to be tactical about when they grab the AI and mitigates the satellite fort strategy since the crew will be tipped off pretty fast once the AI has fallen so theres wont be as much time to turn the AI sat into an autism fort before interlopers show up.

**Comment by ChangelingRain** [comment]
Yeah; it doesn't totally _fix_ the problem, but it means that once you have the AI you need to be building defenses, and won't have nearly as much time to do so, because there's a high chance you're going to get fucking stormed by security.

**Comment by coiax** [comment]
Can they be carded if they die?

**Comment by ChangelingRain** [comment]
No. But they do become unanchored, so Servants can still revive them with vitality matrices.

**Pull request merged**